src/prefect/task_engine.py

```diff
@@ -521,9 +538,23 @@ def initialize_run(
                     self._client = None
 
     def is_running(self) -> bool:
-        if getattr(self, "task_run", None) is None:
+        """Whether or not the engine is currently running a task."""
+        if (task_run := getattr(self, "task_run", None)) is None:
             return False
-        return getattr(self, "task_run").state.is_running()
+        return task_run.state.is_running() or task_run.state.is_scheduled()
+
+    @sync_compatible
+    async def wait_until_ready(self):
```

Comment:
Can you use `run_coro_as_sync` directly in the control loop instead of using `@sync_compatible` as was mentioned here: https://github.com/PrefectHQ/prefect/pull/13785#discussion_r1626867299